src/parquet/schema.cc

```diff
@@ -620,6 +620,15 @@ const ColumnDescriptor* SchemaDescriptor::Column(int i) const {
   return &leaves_[i];
 }
 
+int SchemaDescriptor::ColumnIndex(const NodePtr& node) const {
```

Comment:
It seemed to me like a good idea at first, but unfortunately this creates a bit of a problem of using `shared_ptr` (which `NodePtr` are) as keys to an `unordered_map`. Doing it with std objects seems complicated (seems like boost solved it but don't think we want to convert all std::shared_ptr to boost:shared_ptr).

I can use Node* instead but that might be bug prone.